wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -386,8 +478,228 @@ class PlatformRepo(val walletApplication: WalletApplication) {
-            val dashPayProfile = DashPayProfile.fromDocument(profile)
+            val dashPayProfile = DashPayProfile.fromDocument(profile, username)
             updateDashPayProfile(dashPayProfile!!)
         }
     }
+
+    // contacts
+
+    suspend fun updateContactRequests(userId: String) {
+
+        val userIdList = HashSet<String>()
+
+        // Get all out our contact requests
+        val toContactDocuments = ContactRequests(platform).get(userId, toUserId = false, retrieveAll = true)
+        toContactDocuments.forEach {
+            val toUserId = it.data["toUserId"] as String
+            userIdList.add(toUserId)
+            val privateData = if (it.data.containsKey("privateData"))
+                HashUtils.byteArrayFromString(it.data["privateData"] as String)
+            else null
+            val contactRequest = DashPayContactRequest(it.entropy, it.userId, it.data["toUserId"] as String,
+                    privateData,
+                    HashUtils.byteArrayFromString(it.data["encryptedPublicKey"] as String),
+                    it.data["senderKeyIndex"] as Int,
+                    it.data["recipientKeyIndex"] as Int,
+                    it.data["timestamp"] as Double, false, 0)
+            dashPayContactRequestDaoAsync.insert(contactRequest)
+        }
+        // Get all contact requests where toUserId == userId, the users who have added me
+        val fromContactDocuments = ContactRequests(platform).get(userId, toUserId = true, retrieveAll = true)
+        fromContactDocuments.forEach {
+            userIdList.add(it.userId)
+            val privateData = if (it.data.containsKey("privateData"))
+                HashUtils.byteArrayFromString(it.data["privateData"] as String)
+            else null
+            val contactRequest = DashPayContactRequest(it.entropy, it.userId, it.data["toUserId"] as String,
+                    privateData,
+                    HashUtils.byteArrayFromString(it.data["encryptedPublicKey"] as String),
+                    it.data["senderKeyIndex"] as Int,
+                    it.data["recipientKeyIndex"] as Int,
+                    it.data["timestamp"] as Double, false, 0)
+            dashPayContactRequestDaoAsync.insert(contactRequest)
+        }
+
+        var nameDocuments = HashMap<String, Document>()
+        var profileDocuments = HashMap<String, Document?>()
+
+        for (id in userIdList) {
+            val nameDocument = platform.names.getByUserId(id)
+            nameDocuments[id] = nameDocument[0]
+
+            val profileDocument = profiles.get(id) ?: profiles.createProfileDocument("", "",
+                    "", platform.identities.get(nameDocument[0].userId)!!)
+
+            profileDocuments[id] = profileDocument
+
+            val profile = DashPayProfile.fromDocument(profileDocument, nameDocument[0].data["normalizedLabel"] as String)
+            dashPayProfileDaoAsync.insert(profile!!)
+
+        }
+
+        // lets add more data
```

Comment:
this section adds many contacts to the database.  The userId's and usernames do exist.  This may help with other screens until contacts can actually be added.